addons/project_purchase/models/project_project.py

```diff
@@ -113,7 +113,7 @@ def _get_profitability_items(self, with_action=True):
         profitability_items = super()._get_profitability_items(with_action)
         if self.analytic_account_id:
             purchase_order_lines = self.env['purchase.order.line'].sudo().search_fetch([
-                ('state', 'in', ['purchase', 'done']),
+                ('order_id.invoice_status', '=', 'invoiced'),
```

Comment:
The problem by doing is we could reduce the performance since we will have to do a subquery to check if `invoice_status == 'invoiced'` in the `purchase.order`. 🤔 

We could maybe just check if the `qty_invoiced` is greater than 0? 

We should have something like that:

```py
            purchase_order_lines = self.env['purchase.order.line'].sudo().search_fetch([
                ('state', 'in', ['purchase', 'done']),
                ('analytic_distribution', 'in', self.analytic_account_id.ids),
                ('qty_invoiced', '>', 0),
            ], ['qty_invoiced', 'qty_to_invoice', 'product_uom_qty', 'price_unit', 'currency_id', 'analytic_distribution'])

```
🤷🏼‍♂️ 

(not tested)

What do you think? 🤔

Reply:
I tested and it seems to do the trick :) The performance should be better indeed